pkgs/applications/window-managers/qtile/default.nix

```diff
@@ -1,66 +1,127 @@
-{ lib, fetchFromGitHub, python37Packages, glib, cairo, pango, pkg-config, libxcb, xcbutilcursor }:
+{ lib
+, fetchFromGitHub
+, buildPythonApplication
+, wrapGAppsHook
+, cairo
+, cairocffi
+, dateutil
+, dbus-python
+, glib
+, gobject-introspection
+, libnotify
+, libpulseaudio
+, libxcb
+, mpd2
+, pango
+, pkg-config
+, psutil
+, pygobject3
+, pyxdg
+, setuptools
+, setuptools_scm
+, xcbutilcursor
+, xcffib
 
-let cairocffi-xcffib = python37Packages.cairocffi.override {
+, dbus
+, pytest
+, xcalc
+, xclock
+, xeyes
+, xorgserver
+, xterm
+, xvfb-run
+}:
+let cairocffi-xcffib = cairocffi.override {
     withXcffib = true;
   };
 in
-
-python37Packages.buildPythonApplication rec {
+buildPythonApplication rec {
   name = "qtile-${version}";
-  version = "0.16.0";
+  version = "0.17.0";
 
   src = fetchFromGitHub {
     owner = "qtile";
     repo = "qtile";
     rev = "v${version}";
-    sha256 = "1klv1k9847nyx71sfrhqyl1k51k2w8phqnp2bns4dvbqii7q125l";
+    sha256 = "0n1scjrk9pwjjivpsvp8gx2yqssk79hgsx5d0dlay9kk7bzvsx0w";
   };
 
   patches = [
     ./0001-Substitution-vars-for-absolute-paths.patch
     ./0002-Restore-PATH-and-PYTHONPATH.patch
     ./0003-Restart-executable.patch
+    ./0004-Fix-test-propgated-PYTHONPATH-in-Popen.patch
   ];
 
   postPatch = ''
     substituteInPlace libqtile/core/manager.py --subst-var-by out $out
     substituteInPlace libqtile/pangocffi.py --subst-var-by glib ${glib.out}
     substituteInPlace libqtile/pangocffi.py --subst-var-by pango ${pango.out}
-    substituteInPlace libqtile/backend/x11/xcursors.py --subst-var-by xcb-cursor ${xcbutilcursor.out}
+    substituteInPlace libqtile/backend/x11/xcursors.py --subst-var-by xcb-cursor ${xcbutilcursor}
+    patchShebangs ./scripts/xephyr
   '';
 
   SETUPTOOLS_SCM_PRETEND_VERSION = version;
 
-  nativeBuildInputs = [ pkg-config ];
-  buildInputs = [ glib libxcb cairo pango python37Packages.xcffib ];
-
-  pythonPath = with python37Packages; [
-    xcffib
+  nativeBuildInputs = [ pkg-config wrapGAppsHook ];
```

Comment:
Add lm_sensors

Reply:
Is lm_sensors a checkPhase dependency or a build time tool dependency ?